internal/resources/azure/mysql_flexible_server.go

```diff
@@ -0,0 +1,161 @@
+	productNameRegex := fmt.Sprintf("^Azure Database for MySQL Flexible Server %s\\s+%s", attrs.TierName, attrs.Series)
+
+	return &schema.CostComponent{
+		Name:           fmt.Sprintf("Compute (%s)", r.SKU),
+		Unit:           "hours",
+		UnitMultiplier: decimal.NewFromInt(1),
+		HourlyQuantity: decimalPtr(decimal.NewFromInt(1)),
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("azure"),
+			Region:        strPtr(r.Region),
+			Service:       strPtr("Azure Database for MySQL"),
+			ProductFamily: strPtr("Databases"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "productName", ValueRegex: regexPtr(productNameRegex)},
+				{Key: "skuName", ValueRegex: regexPtr(fmt.Sprintf("^%s$", attrs.SKUName))},
+				{Key: "meterName", ValueRegex: regexPtr(fmt.Sprintf("^%s$", attrs.MeterName))},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			PurchaseOption: strPtr("Consumption"),
+		},
+	}
+}
+
+// storageCostComponent returns a cost component for server's storage.
+func (r *MySQLFlexibleServer) storageCostComponent() *schema.CostComponent {
+	var quantity *decimal.Decimal
+	if r.Storage > 0 {
+		quantity = decimalPtr(decimal.NewFromInt(r.Storage))
+	}
+
+	return &schema.CostComponent{
+		Name:            "Storage",
+		Unit:            "GB",
+		UnitMultiplier:  decimal.NewFromInt(1),
+		MonthlyQuantity: quantity,
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("azure"),
+			Region:        strPtr(r.Region),
+			Service:       strPtr("Azure Database for MySQL"),
+			ProductFamily: strPtr("Databases"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "productName", Value: strPtr("Azure Database for MySQL Flexible Server Storage")},
+				{Key: "meterName", Value: strPtr("Storage Data Stored")},
+			},
+		},
+	}
+}
+
+// iopsCostComponent returns a cost component for additional IOPS.
+func (r *MySQLFlexibleServer) iopsCostComponent() *schema.CostComponent {
+	var quantity *decimal.Decimal
+	if r.IOPS > 0 {
+		quantity = decimalPtr(decimal.NewFromInt(r.IOPS))
+	}
+
+	return &schema.CostComponent{
+		Name:            "Additional IOPS",
+		Unit:            "IOPS",
+		UnitMultiplier:  decimal.NewFromInt(1),
+		MonthlyQuantity: quantity,
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("azure"),
+			Region:        strPtr(r.Region),
+			Service:       strPtr("Azure Database for MySQL"),
+			ProductFamily: strPtr("Databases"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "productName", Value: strPtr("Azure Database for MySQL Flexible Server Storage")},
+				{Key: "skuName", Value: strPtr("Additional IOPS")},
+			},
+		},
+	}
+}
+
+// backupCostComponent returns a cost component for additional backup storage.
+func (r *MySQLFlexibleServer) backupCostComponent() *schema.CostComponent {
+	var quantity *decimal.Decimal
+	if r.AdditionalBackupStorageGB != nil {
+		quantity = decimalPtr(decimal.NewFromFloat(*r.AdditionalBackupStorageGB))
```

Comment:
According to pricing page if geo-redundancy is enabled, the storage is doubled:

> GRS requires twice the amount of storage capacity because it creates a replica. If you have configured the back up storage to be geo-redundant (GRS), the price will be 2x this price. Price per GB remains the same.

Should the usage parameter include that? Or should we rely on [geo_redundant_backup_enabled](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mysql_flexible_server#geo_redundant_backup_enabled) flag and double the storage (and cost) here if it is set to true?

If yes, we'll need to fix it for PostgreSQL Flexible Server as well.

Reply:
Good question. I can see benefits of both.

1. Assuming the usage parameter includes it would probably map better to any cost and usage reports.
2. Doing the doubling ourself might be simpler to the user for modelling.

I'm leaning towards 1 just now because it's less ambiguous, but we'd need to add a comment to the usage file to make it clear that for GRS you should double it yourself. What do you think?

Reply:
Agree, ambiguity is the main problem here. I'd assume users know what they are doing when they set the geo-redundancy flag, so something like maybe enough for now:
```yaml
  azurerm_mysql_flexible_server.my_flexible_server:
    additional_backup_storage_gb: 5000 # Additional consumption of backup storage in GB. This includes the replica storage when geo-redundancy is enabled.
```

Is it clear enough?

Reply:
Maybe this:

> Additional backup storage in GB. If geo-redundancy is enabled, you should set this to twice the required storage capacity.

Reply:
Yep, that's better 👍 . Thanks!

Reply:
Updated, also for PostgreSQL Flexible Server.